docs/blitz/design/actions/actions_list.rst

```diff
@@ -404,6 +404,53 @@ snapshot you can specify to not delete it. See `example` below.
         device: my_device
         delete_snapshot: False #optional, default is True
 
+yang_snapshot
+^^^^^^^^^^^^^
+
+The ``yang_snapshot`` action is another alternative to capture related device
+configuration in a lightweight way. It does not save all device configuration,
+instead, it scans Genie Trigger information and find out ``yang`` actions that
+will change the device configuration before the next ``yang_snapshot_restore``
+action. Note that one ``yang_snapshot`` action can be followed by one or
```

Comment:
Lines 411 - 414 is a bit confusing.  Are you referring to something like this?

"It doesn't actually save device configurations, but instead, it begins recording all yang actions that change the configuration.  When ``yang_snapshot_restore`` is called, all the stored yang action changes are reversed."

Reply:
It does save device configuration, but not all configuration of all YANG models. It picks related YANG models and save their configurations. On top of that, all `yang` actions are tracked.

Reply:
So it does a ``get-config`` of the datastore _**and**_ tracks yang actions?

Reply:
Exactly, it does a `get-config` and tracks `yang` actions.